src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -301,6 +318,23 @@ private FormValidation checkAgentConnectivity(final String host, final String po
         return FormValidation.ok("Success!");
     }
 
+     /**
+     * Gets the StringCredentials object for the given credential ID
+     *
+     * @param credentialId - The Id of the credential to get
+     * @return a StringCredentials object
+     */
+    public StringCredentials getCredentialFromId(String credentialId) {
+        return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                    StringCredentials.class,
+                    Jenkins.get(),
+                    ACL.SYSTEM,
+                    URIRequirementBuilder.fromUri(null).build()),
+                CredentialsMatchers.allOf(CredentialsMatchers.withId(credentialId))
```

Comment:
What happens is `credentialId` is null?

Reply:
Ah I don't think it can be null. Added a null check before calling it in `findSecret`